share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -380,5 +380,58 @@ function(sysbuild_module_call)
       endforeach()
     endif()
   endforeach()
+endfunction()
+
+# Usage:
+#   sysbuild_cache_set(VAR <variable> [APPEND [LIST]] <value>)
```

Comment:
> It is optional

my bad, I had read this:
https://github.com/zephyrproject-rtos/zephyr/blob/9ac601de052a2fd486c083c1995810d84255b51a/share/sysbuild/cmake/modules/sysbuild_extensions.cmake#L415
as `(VARS_APPEND AND NOT VARS_LIST)` which would have made both arguments required if APPEND was used.

> if LIST is not provided, is treated as a string e.g. the result would be "abc" + "eeq" = "abceeq" 

although the intention is good, i'm very concerned with such approach, because if any CMake code does:
```
sysbuild_cache_set(VAR CACHE_STRING APPEND "hello") 
```
you will get an ever growing string on each CMake re-run.
And you cannot string de-duplication safely, like you can do with list items.

Reply:
in that case I will remove appending strings

Reply:
Removed string appending